File: VB.NET/windows/help_error/Backup/help_error/Form1.vb
Public Class Form1

   
    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load


    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If (TextBox1.Text = "") Then
            ErrorProvider1.SetError(TextBox1, "user id should not empty")

        End If
        If (TextBox2.Text = "") Then
            ErrorProvider1.SetError(TextBox2, "pwdshould not empty")

        End If
    End Sub

    Private Sub TextBox1_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TextBox1.TextChanged
        ToolTip1.SetToolTip(TextBox1, "Enter only no")
    End Sub

    Private Sub TextBox2_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TextBox2.TextChanged
        HelpProvider1.SetHelpString(TextBox2, "Enter  pwd")
    End Sub
End Class

File: csharp/Windows/ctrl2/Backup/ctrl2/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;


namespace ctrl2
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("Welcome", "hello");

            
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {
            helpProvider1.SetHelpString(textBox1, "Enter TextBox value");
        }

        private void textBox2_TextChanged(object sender, EventArgs e)
        {
            toolTip1.SetToolTip(textBox2, "Enter only no");
        }

        private void textBox4_TextChanged(object sender, EventArgs e)
        {
            if (textBox1.Text=="") 
                  errorProvider1.SetError(textBox1, "user id should not empty");
            //if (textBox2.Text == "") 
            //      errorProvider1.SetError(textBox2, "pwdshould not empty");
      
        }
    }
}